docs: make courses docs beginner friendly (#1159)
kantord/LibreLingo: docs/courses/skill.md

- `Alternative versions`: An alternative version of the above phrase. (optional)
- `Translation`: Translation of the phrase in target audience's language. e.g. In this case, it's written in English.

**`Mini-dictionary`** has a list of words and meanings. [TODO: Explain where and how it is used]
- `<destination language>`: A list of words in destination language as key and meaning in target audience's language.
- `<target audience's language>`: A list of words in target audience's language as key and meaning in destination language.

<a id="creating-new"></a>
## Creating new skills

Creating new skills is a matter of creating their corresponding YAML file.
That being said, there are a couple things to keep in mind:

- Skill yaml file should be inside a [module](module.md)'s `skills` directory.
- Skills should have an unique `Id`. Such identifiers can be generated using
  [this website](https://www.uuidgenerator.net/version4).
- Before skills will actually show up in the web app, they need to be added to
  a [`module.yaml`](module.md#yaml).

<a id="example-edit"></a>
## Examples of editing a skill

> For example, this is how you'd want to add a new alternative version to the

Reviewer: ```suggestion
This is how you'd add a new alternative version to the
```

phrase "La femme dit bonjour":

```diff
Phrases:
  - Phrase: La femme dit bonjour
    Alternative versions:
      - La femme dit salut
+     - La madame dit salut
    Translation: The woman says hello
    Alternative translations:
      - The woman says hi
```

Keep in mind that when you add a new phrase, or a new version or translation

> to a phrase, it's not automatically updating to the mini-dictionary. So you'd

Reviewer: ```suggestion
to a phrase, it's not automatically updating to the mini-dictionary. So you need to
```

Author: There is a line after this line:
> want to add any new words there:

Making it:
> to a phrase, it's not automatically updating to the mini-dictionary. So you need to
> want to add any new words there:

So we'd have to remove "want to" from the next line:
```
to a phrase, it's not automatically updating to the mini-dictionary. So you need to
add any new words there:
```

I think this is getting a bit confusing. Reading 80 char wrapped lines esp. for grammar related stuff is a bit problematic and hard to get context. I mean just checking all the other review threads and this, it made me look back forth the version before and suggestions.

I copied this part from the old version as is, so the hard wrap remained. Do you think we should unwrap these and make them normal as the rest of the changes in this PR?

Reviewer: > I copied this part from the old version as is, so the hard wrap remained. Do you think we should unwrap these and make them normal as the rest of the changes in this PR?

Or maybe there's a way to have multiline suggestions in GitHub? 

Or I'll just avoid the suggestion feature in such cases

Author: Well, the mixed styling of line wraps in the same file doesn't look nice either. But I guess we can live with it for now.

Reviewer: We can use some auto formatting/linting later to unify the styles. We're using such things for Python and TypeScript/JavaScript. But it doesn't have to be done in this MR

> want to add any new words there:

Reviewer: ```suggestion
add any new words there:
```

Reviewer: Also maybe we should mention that you need to add the new words in both languages

Author: I've added something like this. Resolve this if it's acceptable.
````
Also note, that the same phrase should be added in both the languages under `Mini-dictionary` key. Meaning, if you add a phrase under target language, you should add it under the source language too. For example:

```yaml
Mini-dictionary:
  French:
    - dit: says
    - bonjour:
      - hello
      - hi
...
  English:
    - says: dit
    - hello:
      - bonjour
      - salut
```
````

Reviewer: > Also note, that the same phrase should be added in both the languages under `Mini-dictionary` key. Meaning, if you add a phrase under target language, you should add it under the source language too. For example:

Yes, but I would maybe simplify it:

> Also note, that the new words from each phrase need to be added for both the source and the target language in the `Mini-dictionary`.  For example:

Author: I think a clarification would be helpful to users.

> new words from each phrase

I have these questions appear to me when I read this:
- What does it mean by "new words"?
- New words from phrases from where? Is this referring to phrases under `Phrases` key?

Reviewer: > is this referring to phrases under Phrases key?

yes

> What does it mean by "new words"?

Words that haven't been defined earlier. The system can re-use definitions from other Skills, although duplicates are also allowed, so it's not an issue if the same definition is included twice.

Author: Maybe something like this then...?
```
Also note, that the new words from each phrase listed under `Phrases` key need to be added for both the source and the target language in the `Mini-dictionary`. For example:

...(example code)...

Don't worry about listing the same word more than once in a course. It will not cause any issues if you do so.
```

```diff
Mini-dictionary:
  French:
    - dit: says
+   - madame: lady
    - bonjour:
      - hello
      - hi
    - L'homme: the man
```

<a id="tips"></a>

> ## Tips for creating good skills

Reviewer: I would maybe move this (and all info about creating the content itself) into a new file :thinking: 

And the title would be something like `How to create good course content`

Reviewer: Another tip we could add on that page is

```
Use Words to teach nouns that can be demonstrated easily with pictures, such as "dog", "car", "tree", "city". Do not try to teach verbs, adjectives, etc. using Words. Instead, use them in phrases.
```

Author: Great. This is my suggestion with slight changes:

```
- Use words to teach nouns that can be demonstrated easily with pictures, such as "dog", "car", "tree", "city".
- Do not try to teach verbs, adjectives, etc. using words. Instead, use them in phrases.
```

- Lower-cased "word"
- Divided it in 2 points

# LibreLingo Documentation on Skill

> A **Skill** resides below a module within the course structure. It's the lowest element on the cource hierarchy. It contains data regarding how LibreLingo will ask the questions to the users.

Reviewer: ```suggestion
A **Skill** resides below a Module within the course structure. It's smallest unit in the course hierarchy. It contains data regarding what questions LibreLingo will ask the to the users.
```

Reviewer: Fair enough, I think `smallest element` is a good alternative :+1:

Author: Are you ok if I keep "lowest element"? I think "lowest" goes well with "element". Sorry I forgot to mention that.

Reviewer: ok

> A skill is usually focused on a particular topic on which the questions are asked.

Reviewer: ```suggestion
A skill usually is centered around a specific theme. This could be introducing a new grammar concept or just vocabulary to talk about a certain topic.
```

To get a better understanding, you can read the [course basics](README.md#basics).

---

**Confused?**
[Ask people](https://github.com/kantord/LibreLingo/discussions) on GitHub Discussions.

---

**Table of Contents:**
- [Tree structure](#tree-structure)
- [`(skill_name).yaml`](#yaml)
  - [Data breakdown](#data-breakdown)
- [Creating new skills](#creating-new)
- [Examples of editing a skill](#example-edit)
- [Tips for creating good skills](#tips)

## Tree structure

> A typical tree structure for skills is like this:

Reviewer: ```suggestion
A typical tree structure for skills of a module is as follows:
```

Reviewer: Good point!

```
courses/spanish-from-english/basics/skills/
├── animals.yaml
├── clothes.yaml
├── food.yaml
├── nature.yaml
├── plurals.yaml
├── professions.yaml
├── verb_plurals.yaml
└── verbs.yaml
```

> Here, `animals.yaml`, `clothes.yaml` and `food.yaml` etc. are skills. These skill files are kept inside a `skills` folder. The yaml files contain information about the skill (see below).

Reviewer: YAML should be capitalized everywhere

`yaml` :arrow_right:  `YAML`
